src/libpsl-native/src/getcurrentthreadid.cpp

```diff
@@ -17,6 +21,9 @@ pid_t GetCurrentThreadId()
     uint64_t tid64;
     pthread_threadid_np(NULL, &tid64);
     tid = (pid_t)tid64;
+#elif defined(__FreeBSD__)
+    tid = pthread_getthreadid_np();
+    return (int)tid;
```

Comment:
This cast is superfluous - this function returns `pid_t`, which is the type of `tid`. Probably want in the line above `tid = (pid_t)pthread_getthreadid_np();`, or just avoid explicit casts altogether.

Reply:
This one...I tried my best here to follow what is done under dotNET for `pthread_getthreadid_np()`  but found it [inconsistent](https://github.com/search?q=repo%3Adotnet%2Fruntime%20pthread_getthreadid_np&type=code). I can make changes both to this PR and open PRs in runtime to correct the usage it in order to avoid any information loss due to casting.